lib/DxilRootSignature/DxilRootSignatureSerializer.cpp

```diff
@@ -265,13 +265,14 @@ void SerializeRootSignatureTemplate(_In_ const T_ROOT_SIGNATURE_DESC* pRootSigna
   memcpy(pSS, pRS->pStaticSamplers, StaticSamplerSize);
 
   // Create the result blob.
-  CDxcMallocHeapPtr<char> bytes(DxcGetThreadMallocNoRef());
+  IMalloc* pMallocNoRef = DxcGetThreadMallocNoRef();
+  CDxcMallocHeapPtr<char> bytes(pMallocNoRef);
   CComPtr<IDxcBlob> pBlob;
   unsigned cb = Serializer.GetSize();
   DXASSERT_NOMSG((cb & 0x3) == 0);
   IFTBOOL(bytes.Allocate(cb), E_OUTOFMEMORY);
   IFT(Serializer.Compact(bytes.m_pData, cb));
-  IFT(DxcCreateBlobOnHeap(bytes.m_pData, cb, ppBlob));
+  IFT(DxcCreateBlobOnMalloc(bytes.m_pData, pMallocNoRef, cb, ppBlob));
```

Comment:
Could provide an access for the allocator in `CDxcMallocHeapPtr` (or make it public like the data). That way you can pass the data and the allocator from teh same place (e.g. using `bytes.m_pAllocator`)